lib/facility_factory.rb

```diff
@@ -0,0 +1,70 @@
+class FacilityFactory
+  # This is a helper method for address_parser 
+  def or_address_parser(facility)
+    address_1 = JSON.parse(facility[:location_1][:human_address]).values # returns an array with address line 1, city, state, zip
+    #some addresses have line 2
+    if facility.has_key?(:location_2)
+      address_2 = JSON.parse(facility[:location_2][:human_address]).values[0] # returns the address line 2 value when present
+      address_1.insert(1,address_2) # adds the line 2 to the address array
+    end
+    address_1.join(' ')
+  end
+  
+  # This is a helper method for address_parser
+  def ny_address_parser(facility)
+    [titleize(facility[:street_address_line_1]),
+    facility[:street_address_line_2], # if line 2 not present, .compact removes nil
+    titleize(facility[:city]),
+    facility[:state], facility[:zip_code]
+    ].compact.join(" ")
+  end
+  
+  # This is a helper method for address_parser
+  def mo_address_parser(facility)
+    [titleize(facility[:address1]),
+    facility[:address2], # if line 2 not present, .compact removes nil
+    titleize(facility[:city]),
+    facility[:state],
+    facility[:zipcode]
+  ].compact.join(" ")
+  end
+  
+  # This is a helper function to capitalize the first character of each word
+  def titleize(string)
+    string.split(' ').map(&:capitalize).join(" ")
+  end
+  
+  def address_parser(facility)
+    if facility.has_key?(:location_1)
+      or_address_parser(facility)
+    elsif facility.has_key?(:street_address_line_1)
+      ny_address_parser(facility)
+    elsif facility.has_key?(:address1)
+      mo_address_parser(facility)
+    end
+  end
```

Comment:
YES! The addresses were definitely a tricky part of parsing each state's data. Great idea to have a method that determines how to format the addresses for you.

Reply:
We'll talk on Friday about Modules. This could be a fun time to pull out all methods that deal with addresses into a module, mainly for organization purposes.